lua/starfall/libs_sh/vectors.lua

```diff
@@ -388,26 +388,39 @@ function vec_methods:normalize()
 	self[3] = self[3] / len
 end
 
+local math_sin, math_cos = math.sin, math.cos
+local deg2rad = math.pi/180
+
 --- Rotate the vector by Angle b.
 -- Self-Modifies. Does not return anything
 -- @param Angle b Angle to rotate by.
 function vec_methods:rotate(b)
-	local vec = unwrap(self)
-	vec:Rotate(aunwrap(b))
+	checktype(b, ang_meta, 2)
+	local y, p, r = b.yaw * deg2rad, b.pitch * deg2rad, b.roll * deg2rad
```

Comment:
Now that it's using the SF Angle object, you should use `1, 2, 3` keys, that will avoid the `__index` lookup.